hypothesis-python/src/hypothesis/stateful.py

```diff
@@ -323,6 +325,43 @@ def do_draw(self, data):
         return machine.names_to_values[reference.name]
 
 
+def _convert_targets(targets, target):
+    """Single validator and convertor for target arguments."""
+    if target is not None:
+        if targets:
+            note_deprecation(
+                'Passing both targets=%r and target=%r is redundant, and '
+                'will become an error in a future version of Hypothesis.  '
+                'Just pass targets=%r instead.'
+                % (targets, target, tuple(targets) + (target,))
+            )
+        targets = tuple(targets) + (target,)
+
+    converted_targets = []
+    for t in targets:
+        if isinstance(t, string_types):
+            note_deprecation(
+                'Got %r as a target, but passing the name of a Bundle is '
+                'deprecated - please pass the Bundle directly.' % (t,)
+            )
+        elif not isinstance(t, Bundle):
+            msg = 'Got invalid target %r of type %r, but all targets must ' \
+                'be either a Bundle or the name of a Bundle.'
+            if isinstance(t, OneOfStrategy):
```

Comment:
I wonder how common this particular mistake really is. I don't object to special casing it, I'm just curious.

Reply:
We know that at least one user thought this should work, so there are probably many more who haven't reported anything and IMO it's worth a message that explicitly describes the actual model.  Legibility is especially important here where the domain is complex and our docs are relatively weak!

Reply:
Yeah, like I said, I don't at all object to the special casing, I'm just genuinely curious :-)